libraries/resource_providers/api500/c7000/volume_provider.rb

```diff
@@ -0,0 +1,94 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API500
+    module C7000
+      # Volume API500 C7000 provider
+      class VolumeProvider < API300::C7000::VolumeProvider
+        def load_resource_with_associated_resources
+          if @new_resource.volume_template
+            load_volume_template
+          else
+            validate_required_properties(:storage_system, :storage_pool)
+            storage_system_uri = load_storage_system
+            @item.set_storage_pool(resource_named(:StoragePool).new(@item.client, name: @new_resource.storage_pool, storageSystemUri: storage_system_uri))
+            @item.set_snapshot_pool(resource_named(:StoragePool).new(@item.client, name: @new_resource.snapshot_pool, storageSystemUri: storage_system_uri)) if @new_resource.snapshot_pool
+          end
+          @item['properties']['name'] = @new_resource.name
+          @item.exists? ? @item.data.delete('properties') : set_properties
+        end
+
+        def load_storage_system
```

Comment:
Any point to having this one-liner method?

Reply:
did not drop this method as requested on provider 200 because I call it in more than one location, so I thought it would be best to keep it for a single point of maintenance

Reply:
Yeah, but we end up calling the method in two places, instead of that one line of code in two places. Not really a difference worth having a method in this case.

As mentioned in conversation, if this specific load were to be changed in future versions, there *is* a definite advantage to have it in a method, since then we'd just need to overwrite this method instead of the ones calling it. 

However, since that does not appear to be the case at the moment, at least that we know of... It is not adding anything by being a separate method, so I'd suggest dropping it.